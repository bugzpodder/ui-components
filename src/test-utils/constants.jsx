// @flow
import React from "react";
import { EDC, PIPELINE } from "@grail/lib";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL = "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<?string>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map().set(EDC, "edc.com").set(PIPELINE, "pipeline.com");
