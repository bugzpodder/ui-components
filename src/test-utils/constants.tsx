import React from "react";

export const COUNTRY_NAMES = [
  "Afghanistan",
  "Akrotiri",
  "Albania",
  "Algeria",
  "American Samoa",
  "Andorra",
  "Angola",
  "Anguilla",
  "Antarctica",
  "Antigua and Barbuda",
  "Argentina",
  "Armenia",
  "Aruba",
  "Ashmore and Cartier Islands",
  "Australia",
  "Austria",
  "Azerbaijan",
  "Bahamas, The",
  "Bahrain",
  "Bangladesh",
  "Barbados",
  "Bassas da India",
  "Belarus",
  "Belgium",
  "Belize",
  "Benin",
  "Bermuda",
  "Bhutan",
  "Bolivia",
  "Bosnia and Herzegovina",
  "Botswana",
  "Bouvet Island",
  "Brazil",
  "British Indian Ocean Territory",
  "British Virgin Islands",
  "Brunei",
  "Bulgaria",
  "Burkina Faso",
  "Burma",
  "Burundi",
  "Cambodia",
  "Cameroon",
  "Canada",
  "Cape Verde",
  "Cayman Islands",
  "Central African Republic",
  "Chad",
  "Chile",
  "China",
  "Christmas Island",
  "Clipperton Island",
  "Cocos (Keeling) Islands",
  "Colombia",
  "Comoros",
  "Congo, Democratic Republic of the",
  "Congo, Republic of the",
  "Cook Islands",
  "Coral Sea Islands",
  "Costa Rica",
  "Cote d'Ivoire",
  "Croatia",
  "Cuba",
  "Cyprus",
  "Czech Republic",
  "Denmark",
  "Dhekelia",
  "Djibouti",
  "Dominica",
  "Dominican Republic",
  "Ecuador",
  "Egypt",
  "El Salvador",
  "Equatorial Guinea",
  "Eritrea",
  "Estonia",
  "Ethiopia",
  "Europa Island",
  "Falkland Islands (Islas Malvinas)",
  "Faroe Islands",
  "Fiji",
  "Finland",
  "France",
  "French Guiana",
  "French Polynesia",
  "French Southern and Antarctic Lands",
  "Gabon",
  "Gambia, The",
  "Gaza Strip",
  "Georgia",
  "Germany",
  "Ghana",
  "Gibraltar",
  "Glorioso Islands",
  "Greece",
  "Greenland",
  "Grenada",
  "Guadeloupe",
  "Guam",
  "Guatemala",
  "Guernsey",
  "Guinea",
  "Guinea-Bissau",
  "Guyana",
  "Haiti",
  "Heard Island and McDonald Islands",
  "Holy See (Vatican City)",
  "Honduras",
  "Hong Kong",
  "Hungary",
  "Iceland",
  "India",
  "Indonesia",
  "Iran",
  "Iraq",
  "Ireland",
  "Isle of Man",
  "Israel",
  "Italy",
  "Jamaica",
  "Jan Mayen",
  "Japan",
  "Jersey",
  "Jordan",
  "Juan de Nova Island",
  "Kazakhstan",
  "Kenya",
  "Kiribati",
  "Korea, North",
  "Korea, South",
  "Kuwait",
  "Kyrgyzstan",
  "Laos",
  "Latvia",
  "Lebanon",
  "Lesotho",
  "Liberia",
  "Libya",
  "Liechtenstein",
  "Lithuania",
  "Luxembourg",
  "Macau",
  "Macedonia",
  "Madagascar",
  "Malawi",
  "Malaysia",
  "Maldives",
  "Mali",
  "Malta",
  "Marshall Islands",
  "Martinique",
  "Mauritania",
  "Mauritius",
  "Mayotte",
  "Mexico",
  "Micronesia, Federated States of",
  "Moldova",
  "Monaco",
  "Mongolia",
  "Montserrat",
  "Morocco",
  "Mozambique",
  "Namibia",
  "Nauru",
  "Navassa Island",
  "Nepal",
  "Netherlands",
  "Netherlands Antilles",
  "New Caledonia",
  "New Zealand",
  "Nicaragua",
  "Niger",
  "Nigeria",
  "Niue",
  "Norfolk Island",
  "Northern Mariana Islands",
  "Norway",
  "Oman",
  "Pakistan",
  "Palau",
  "Panama",
  "Papua New Guinea",
  "Paracel Islands",
  "Paraguay",
  "Peru",
  "Philippines",
  "Pitcairn Islands",
  "Poland",
  "Portugal",
  "Puerto Rico",
  "Qatar",
  "Reunion",
  "Romania",
  "Russia",
  "Rwanda",
  "Saint Helena",
  "Saint Kitts and Nevis",
  "Saint Lucia",
  "Saint Pierre and Miquelon",
  "Saint Vincent and the Grenadines",
  "Samoa",
  "San Marino",
  "Sao Tome and Principe",
  "Saudi Arabia",
  "Senegal",
  "Serbia and Montenegro",
  "Seychelles",
  "Sierra Leone",
  "Singapore",
  "Slovakia",
  "Slovenia",
  "Solomon Islands",
  "Somalia",
  "South Africa",
  "South Georgia and the South Sandwich Islands",
  "Spain",
  "Spratly Islands",
  "Sri Lanka",
  "Sudan",
  "Suriname",
  "Svalbard",
  "Swaziland",
  "Sweden",
  "Switzerland",
  "Syria",
  "Taiwan",
  "Tajikistan",
  "Tanzania",
  "Thailand",
  "Timor-Leste",
  "Togo",
  "Tokelau",
  "Tonga",
  "Trinidad and Tobago",
  "Tromelin Island",
  "Tunisia",
  "Turkey",
  "Turkmenistan",
  "Turks and Caicos Islands",
  "Tuvalu",
  "Uganda",
  "Ukraine",
  "United Arab Emirates",
  "United Kingdom",
  "United States #1",
  "Uruguay",
  "Uzbekistan",
  "Vanuatu",
  "Venezuela",
  "Vietnam",
  "Virgin Islands",
  "Wake Island",
  "Wallis and Futuna",
  "West Bank",
  "Western Sahara",
  "Yemen",
  "Zambia",
  "Zimbabwe",
];

export const COUNTRIES: Array<{
  label: string;
  value: string;
}> = COUNTRY_NAMES.map(name => ({ label: name })).map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL =
  "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  {
    word: "Pneumonoultramicroscopicsilicovolcanoconiosis",
    origin: "General English",
  },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct
        trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: (
      <div>
        Over 2500 rounds had been fired with the media claiming that only a few
        emus had been killed.
      </div>
    ),
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: (
      <div>
        Sir George Pearce wages war on the local population of emus in
        Australia.
      </div>
    ),
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<string | null>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map()
  .set("EDC", "edc.com")
  .set("pipeline", "pipeline.com");
