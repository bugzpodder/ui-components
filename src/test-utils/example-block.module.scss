.container {
  margin: 1rem 0;

  .header {
    font-family: monospace;
  }

  .text {
    background: #f6f8fa;
    font-size: 1rem;
    padding: .5rem;
  }
}