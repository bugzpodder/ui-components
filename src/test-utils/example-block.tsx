import React from "react";
import styles from "./example-block.module.scss";

type Props = {
  content: any;
  strongHeader: string;
  helperText?: string;
};

export const ExampleBlock = (props: Props) => {
  const { strongHeader, helperText = "", content } = props;
  return (
    <div className={styles.container}>
      <pre data-testid="example" className={styles.text}>
        <strong>{strongHeader}</strong>
        {helperText}= {JSON.stringify(content, null, 2)}
      </pre>
    </div>
  );
};
