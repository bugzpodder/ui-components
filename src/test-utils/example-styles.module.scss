.container {
  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.spinnerContainer {
  position: relative;

  .buttonContainer {
    display: inline-block;
      position: relative;
  }

  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.avatarContainer {
  .inLine {
    display: inline-block;
  }
}

.selectContainer {
  .commonSelect {
    max-width: 350px;
  }
}

.pagedTable {
  height: 600px;
}
