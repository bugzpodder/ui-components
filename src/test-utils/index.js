// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
