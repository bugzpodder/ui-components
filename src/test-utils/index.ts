export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  COUNTRY_NAMES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
  TEST_EXTERNAL_DOMAINS,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
