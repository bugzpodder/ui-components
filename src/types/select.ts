// Suggestion object must include the following keys, but may also include others.
import { ReactNode } from "react";

export type CommonSelectOption = {
  label: ReactNode;
  value: string;
  [x0: string]: any;
};

export type Suggestion = {
  label: string;
  value: string;
  [x0: string]: any;
};
