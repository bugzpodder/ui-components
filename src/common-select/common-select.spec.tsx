import React, { useState } from "react";
import mockConsole from "jest-mock-console";
import { COUNTRIES, TestWrapper } from "../test-utils";
import { CommonSelect } from ".";
import { CommonSelectOption } from "../types/select";
import { cleanup, fireEvent, render } from "@testing-library/react";

afterEach(cleanup);

/** ******* Sync ******* */

const TestCommonSelect = (props: any) => {
  const {
    mockOnChange,
    label,
    "data-testid": dataTestId,
    margin,
    variant,
  } = props;
  const [value, setValue] = useState<CommonSelectOption>({
    label: null,
    value: null,
  });
  const setChange = (value): void => {
    mockOnChange(value);
    setValue(value);
  };
  return (
    <TestWrapper>
      <CommonSelect
        label={label}
        data-testid={dataTestId}
        options={COUNTRIES}
        id="test"
        helperText="some helper text"
        value={value}
        formatOption={datum => (
          <div>
            <div>{datum.label}</div>
            <div>{datum.value}</div>
          </div>
        )}
        onChange={setChange}
        margin={margin}
        variant={variant}
      />
    </TestWrapper>
  );
};

test("render and test CommonSelect", () => {
  const mockOnChange = jest.fn(result => result);
  const { getByText, getByTestId } = render(
    <TestCommonSelect mockOnChange={mockOnChange} />,
  );
  fireEvent.click(getByTestId("common-select-input"));
  // Check document.
  // expect(container).toMatchSnapshot();
  COUNTRIES.forEach(country => {
    expect(getByTestId(country.value)).toBeInTheDocument();
  });
  expect(getByTestId("common-select-helper-text")).toBeInTheDocument();
  expect(getByTestId("common-select-helper-text")).toHaveTextContent(
    "some helper text",
  );

  // Select an item.
  const expectedResult = { label: "Algeria", value: "ALGERIA" };
  fireEvent.click(getByText("Algeria"));
  expect(mockOnChange).toHaveBeenCalledTimes(1);
  expect(mockOnChange.mock.results[0].value).toEqual(expectedResult);
  // TODO(nsawas): Check input value here and after.

  // Remove item.
  fireEvent.click(getByTestId("common-select-close-icon"));
  expect(mockOnChange).toHaveBeenCalledTimes(2);
  expect(mockOnChange.mock.results[1].value).toEqual({});
});

test("CommonSelect label", () => {
  const label = "Country";
  const { getByTestId } = render(<TestCommonSelect label={label} />);
  // TODO (yzhao): adding a label causes indeterminate for attribute on label.
  // expect(container).toMatchSnapshot();
  expect(getByTestId("common-select-input-label")).toHaveTextContent(label);
});

test("CommonSelect variant", () => {
  const { container } = render(<TestCommonSelect variant="filled" />);
  // Check document.
  expect(container).toMatchSnapshot();
});

test("CommonSelect margin", () => {
  const { container } = render(<TestCommonSelect margin="normal" />);
  // Check document.
  expect(container).toMatchSnapshot();
});

test("test disabled CommonSelect", () => {
  const mockOnChange = jest.fn();
  const { container } = render(
    <TestWrapper>
      <CommonSelect
        isDisabled
        options={COUNTRIES}
        onChange={mockOnChange}
        helperText="disabled"
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
  // TODO(nsawas): Test that input has attribute disabled="", and that mockOnChange does not get called.
});

/** ******** Creatable ****** */

test("render creatable CommonSelect", () => {
  const mockOnChange = jest.fn();
  const { container } = render(
    <TestWrapper>
      <CommonSelect
        selectType="creatable"
        inputValue="test"
        onChange={mockOnChange}
      />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
  // TODO(nsawas): Test creatable functionality.
});

/** ******* Async ******* */

const TestAsyncCommonSelect = (props: any) => {
  const { value, inputValue, initialMessage } = props;
  const filterCountries = (inputValue): Promise<any[]> => {
    return new Promise(resolve => {
      const newCountries = COUNTRIES.filter(country =>
        inputValue
          ? country.label.toLowerCase().includes(inputValue.toLowerCase())
          : true,
      ).slice(0, 5);
      setTimeout(() => resolve(newCountries), 500);
    });
  };
  return (
    <TestWrapper>
      <CommonSelect
        selectType="async"
        value={value}
        inputValue={inputValue}
        initialMessage={initialMessage}
        loadOptions={filterCountries}
        onChange={() => {}}
      />
    </TestWrapper>
  );
};

test("async CommonSelect with initial and no options messages", () => {
  const { getByTestId, rerender } = render(<TestAsyncCommonSelect />);
  fireEvent.click(getByTestId("common-select-input"));
  expect(getByTestId("common-select-no-options-message")).toBeInTheDocument();
  expect(getByTestId("common-select-no-options-message")).toHaveTextContent(
    "Begin Typing...",
  );
  rerender(<TestAsyncCommonSelect initialMessage="test custom message" />);
  fireEvent.click(getByTestId("common-select-input"));
  expect(getByTestId("common-select-no-options-message")).toHaveTextContent(
    "test custom message",
  );
  rerender(<TestAsyncCommonSelect inputValue="test" />);
  fireEvent.click(getByTestId("common-select-input"));
  expect(getByTestId("common-select-no-options-message")).toHaveTextContent(
    "No results found",
  );
  // TODO (yzhao): adding a label causes indeterminate aria-label attribute.
  // expect(container).toMatchSnapshot();
});

/** ******* Read-Only ******* */

const TestReadOnlyCommonSelect = (props: any) => {
  const { value, mockOnChange } = props;
  return (
    <TestWrapper>
      <CommonSelect
        readOnly
        options={COUNTRIES}
        value={value}
        onChange={mockOnChange}
      />
    </TestWrapper>
  );
};

test("render read-only CommonSelect with value", () => {
  const mockOnChange = jest.fn();
  const value = { label: "Algeria", value: "ALGERIA" };
  const { container, getByTestId, queryByTestId, rerender } = render(
    <TestReadOnlyCommonSelect value={value} mockOnChange={mockOnChange} />,
  );
  expect(container).toMatchSnapshot();
  expect(queryByTestId("common-select")).not.toBeInTheDocument();
  expect(getByTestId("readonly-text-field")).toBeInTheDocument();
  expect(getByTestId("readonly-text-field")).toHaveTextContent("Algeria");
  rerender(<TestReadOnlyCommonSelect value={{}} mockOnChange={mockOnChange} />);
  expect(getByTestId("readonly-text-field")).toHaveTextContent("-");
});

/** ******* Fail Cases ******* */

test("throw options error", () => {
  mockConsole();
  expect(() => render(<CommonSelect />)).toThrowError();
  expect(console.error).toHaveBeenCalled();
});

test("throw loadOptions error", () => {
  mockConsole();
  expect(() => render(<CommonSelect selectType="async" />)).toThrowError();
  expect(console.error).toHaveBeenCalled();
});
