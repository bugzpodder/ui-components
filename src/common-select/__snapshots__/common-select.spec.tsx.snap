// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonSelect margin 1`] = `
<div>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root autocomplete MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
    role="combobox"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot inputRoot MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      >
        <input
          aria-autocomplete="list"
          aria-describedby="test-helper-text"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiInput-input MuiAutocomplete-input input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
          data-testid="common-select-input"
          id="test-select-input"
          placeholder=""
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
            tabindex="-1"
            title="Clear"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                data-testid="common-select-close-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <p
        class="MuiFormHelperText-root"
        data-testid="common-select-helper-text"
        id="test-helper-text"
      >
        some helper text
      </p>
    </div>
  </div>
</div>
`;

exports[`CommonSelect variant 1`] = `
<div>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root autocomplete MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
    role="combobox"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiAutocomplete-inputRoot inputRoot MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
      >
        <input
          aria-autocomplete="list"
          aria-describedby="test-helper-text"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiFilledInput-input MuiAutocomplete-input input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
          data-testid="common-select-input"
          id="test-select-input"
          placeholder=""
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
            tabindex="-1"
            title="Clear"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                data-testid="common-select-close-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained"
        data-testid="common-select-helper-text"
        id="test-helper-text"
      >
        some helper text
      </p>
    </div>
  </div>
</div>
`;

exports[`render creatable CommonSelect 1`] = `
<div>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root autocomplete MuiAutocomplete-hasClearIcon"
    role="combobox"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot inputRoot MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      >
        <input
          aria-autocomplete="list"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiInput-input MuiAutocomplete-input input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
          data-testid="common-select-input"
          id="select-input"
          placeholder=""
          spellcheck="false"
          type="text"
          value="test"
        />
        <div
          class="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
            tabindex="-1"
            title="Clear"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                data-testid="common-select-close-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`render read-only CommonSelect with value 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    Algeria
  </span>
</div>
`;

exports[`test disabled CommonSelect 1`] = `
<div>
  <div
    aria-expanded="false"
    class="MuiAutocomplete-root autocomplete MuiAutocomplete-hasPopupIcon"
    role="combobox"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot inputRoot Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      >
        <input
          aria-autocomplete="list"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiInput-input MuiAutocomplete-input input MuiAutocomplete-inputFocused Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd"
          data-testid="common-select-input"
          disabled=""
          id="select-input-disabled"
          placeholder=""
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Open"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <p
        class="MuiFormHelperText-root Mui-disabled"
        data-testid="common-select-helper-text"
      >
        disabled
      </p>
    </div>
  </div>
</div>
`;
